Check for `zstash` directory in `tests/base.py`

In testing #279, I'm able to log into globus now with @forsyth2's edits to by `.zstash.ini`.  I now get:
```
$ python -m unittest tests/test_globus.py
****************************************
testLsGlobus
****************************************
*** Setup ***
****************************************
Current_directory=/qfs/people/asay932/e3sm_work/zstash/main does not end in zstash
****************************************
Current directory=/qfs/people/asay932/e3sm_work/zstash/main
New current directory=/qfs/people/asay932/e3sm_work/zstash/main
Removing test files, both locally and at the HPSS repo
Globus delete FAILED
F
======================================================================
FAIL: testLs (tests.test_globus.TestGlobus)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/qfs/people/asay932/e3sm_work/zstash/main/tests/test_globus.py", line 183, in testLs
    self.helperLsGlobus(
  File "/qfs/people/asay932/e3sm_work/zstash/main/tests/test_globus.py", line 169, in helperLsGlobus
    use_hpss = self.setupDirs(test_name)
  File "/qfs/people/asay932/e3sm_work/zstash/main/tests/base.py", line 198, in setupDirs
    self.assertWorkspace()
  File "/qfs/people/asay932/e3sm_work/zstash/main/tests/base.py", line 150, in assertWorkspace
    self.stop(
  File "/qfs/people/asay932/e3sm_work/zstash/main/tests/base.py", line 140, in stop
    self.fail(error_message)
AssertionError: Current_directory=/qfs/people/asay932/e3sm_work/zstash/main does not end in zstash
```
This is caused by this check:
https://github.com/E3SM-Project/zstash/blob/main/tests/base.py#L148-L152

To me, expecting that you are in a directory called `zstash` is not reasonable either for making sure you are in the right place.  First, someone could be in the `zstash` subdirectory for the actual package and this test would pass (incorrectly) or they could be in any other directory named `zstash`.  Second, I can clone the repo with any name I like, and I called it `main` so I can put worktrees next to it with other branch names (my usual workflow).  So the test incorrectly fails for me.

I would instead check for a `zstash` subdirectory and a `tests` subdirectory within the current directory. That would be significantly more reliable.

--- a/tests/base.py
+++ b/tests/base.py
@@ -147,12 +147,14 @@ class TestZstash(unittest.TestCase):
         """
         Assert that the code is being run from the correct directory.
         """
-        actual = os.getcwd()
-        expected = "zstash"
-        if not actual.endswith(expected):
-            self.stop(
-                "Current_directory={} does not end in {}".format(actual, expected)
-            )
+        # To run the tests, we should be at the top of the repository.
+        ls_results = os.listdir()
+        if ("zstash" in ls_results) and (os.path.isdir("zstash")):
+            if ("tests" in ls_results) and (os.path.isdir("tests")):
+                return
+        self.stop(
+            f"Current directory={os.getcwd()} does not contain zstash and tests directories"
+        )
 
     def assertEqualOrStop(self, actual, expected):
         """

